packages/@lwc/integration-karma/test/rendering/noframes/index.spec.js

```diff
@@ -0,0 +1,24 @@
+import { createElement } from 'lwc';
+import Component from 'x/component';
+
+it('renders noframes correctly - W-16784305', async () => {
+    const elm = createElement('x-component', { is: Component });
+    document.body.appendChild(elm);
+
+    await Promise.resolve();
+
+    const {
+        shadowRoot: { childNodes },
+    } = elm;
+
+    expect(childNodes.length).toBe(1);
+    expect(childNodes[0].tagName).toBe('SECTION');
+    expect(childNodes[0].childNodes.length).toBe(1);
+    expect(childNodes[0].childNodes[0].tagName).toBe('NOFRAMES');
+    expect(childNodes[0].childNodes[0].childNodes.length).toBe(1);
+    expect(childNodes[0].childNodes[0].childNodes[0].tagName).toBe('DIV');
+    expect(childNodes[0].childNodes[0].childNodes[0].childNodes.length).toBe(0);
+    expect(childNodes[0].childNodes[0].childNodes[0].getAttribute('class')).toBe(
```

Comment:
Gross.

Reply:
Avoiding innerHTML/outerHTML because it doesn't work properly in synthetic shadow DOM.